edx_event_bus_kafka/consumer/event_consumer.py

```diff
@@ -35,121 +35,136 @@
 EVENT_TYPE_HEADER = "ce_type"
 
 
-def create_consumer(group_id) -> DeserializingConsumer:
+class KafkaEventConsumer:
     """
-    Create a consumer for SESSION_LOGIN_COMPLETED events.
+    Constructs consumers for the given topic, group, and signal, and then
+    emits signals.
 
-    Note: Still needs to be expanded to cover arbitrary events.
-
-    Arguments:
-        group_id: id of the consumer group this consumer will be part of
+    Can also consume messages indefinitely off the queue.
     """
 
-    # TODO (EventBus): Deduplicate settings/client construction against producer code.
-    KAFKA_SCHEMA_REGISTRY_CONFIG = {
-        'url': settings.SCHEMA_REGISTRY_URL,
-        'basic.auth.user.info': f"{getattr(settings, 'SCHEMA_REGISTRY_API_KEY', '')}"
-                                f":{getattr(settings, 'SCHEMA_REGISTRY_API_SECRET', '')}",
-    }
-
-    schema_registry_client = SchemaRegistryClient(KAFKA_SCHEMA_REGISTRY_CONFIG)
-
-    # TODO (EventBus):
-    # 1. Reevaluate if all consumers should listen for the earliest unprocessed offset (auto.offset.reset)
-    # 2. Ensure the signal used in the signal_deserializer is the same one sent over in the message header
-
-    signal_deserializer = AvroSignalDeserializer(SESSION_LOGIN_COMPLETED)
-
-    def inner_from_dict(event_data_dict, ctx=None):  # pylint: disable=unused-argument
-        return signal_deserializer.from_dict(event_data_dict)
-
-    consumer_config = {
-        'bootstrap.servers': settings.KAFKA_BOOTSTRAP_SERVERS,
-        'group.id': group_id,
-        'key.deserializer': StringDeserializer('utf-8'),
-        'value.deserializer': AvroDeserializer(schema_str=signal_deserializer.schema_string(),
-                                               schema_registry_client=schema_registry_client,
-                                               from_dict=inner_from_dict),
-        'auto.offset.reset': 'earliest'
-    }
+    def __init__(self, topic, group_id, signal):
+        self.topic = topic
+        self.group_id = group_id
+        self.signal = signal
 
-    if getattr(settings, 'KAFKA_API_KEY', None) and getattr(settings, 'KAFKA_API_SECRET', None):
-        consumer_config.update({
-            'sasl.mechanism': 'PLAIN',
-            'security.protocol': 'SASL_SSL',
-            'sasl.username': settings.KAFKA_API_KEY,
-            'sasl.password': settings.KAFKA_API_SECRET,
-        })
+    def create_consumer(self) -> DeserializingConsumer:
```

Comment:
[blocker][unless you have a strong argument]
I think this should be part of the init and we should just store the consumer (maybe keeping this as a private method). Otherwise the class feels more of a KafkaEventConsumerFactory or something.